root/static/scripts/common/components/CDTocReleaseListRow.js

```diff
@@ -0,0 +1,181 @@
+const CDTocReleaseListRow = ({
+  associatedMedium,
+  cdTocTrackCount,
+  countInReleaseGroup,
+  release,
+  showArtists = false,
+  wasMbidSearch = false,
+}: Props): React.Element<typeof React.Fragment> => {
+  const $c = React.useContext(SanitizedCatalystContext);
+
+  const [hidden, setHidden] = React.useState<boolean>(true);
+
+  function onButtonClick(event: SyntheticMouseEvent<HTMLAnchorElement>) {
+    event.preventDefault();
+    hidden ? setHidden(false) : setHidden(true);
+  }
+
+  const attachableMediums = release.mediums.filter(medium => (
+      medium.cdtoc_track_count === cdTocTrackCount
+    ));
+
+  return (
+    <>
+      <tr className={loopParity(countInReleaseGroup)}>
+        <td colSpan="2">
+          <EntityLink entity={release} />
+        </td>
+        {showArtists ? (
+          <td>
+            <ArtistCreditLink artistCredit={release.artistCredit} />
+          </td>
+        ) : null}
+        <td>
+          <ReleaseEvents events={release.events} />
+        </td>
+        <td>
+          <ReleaseLabelList labels={release.labels} />
+        </td>
+        <td>
+          <ReleaseCatnoList labels={release.labels} />
+        </td>
+        <td className="barcode-cell">
+          {formatBarcode(release.barcode)}
+        </td>
+        {$c.session?.tport == null ? null : (
+          <td>
+            <TaggerIcon entityType="release" gid={release.gid} />
+          </td>
+        )}
+      </tr>
+      {attachableMediums ? attachableMediums.map((medium, index) => {
+        const cdTocAlreadyAttached =
+          Boolean(associatedMedium) && (medium.id === associatedMedium);
+        const cannotHaveDiscIds = !medium.may_have_discids;
+        const hasLoadedTracks = Boolean(medium.tracks);
+
+        return (
+          <React.Fragment key={index}>
+            <tr className={loopParity(countInReleaseGroup)}>
+              <td className="pos" />
+              <td>
+                <label>
+                  {cdTocAlreadyAttached ? (
+                    <div
+                      className="cannot-attach-discid icon img"
+                      title={l(
+                        'This CDTOC is already attached to this medium.',
+                      )}
+                    />
+                  ) : cannotHaveDiscIds ? (
+                    <div
+                      className="cannot-attach-discid icon img"
+                      title={l(
+                        'This medium format cannot have a disc ID attached.',
+                      )}
+                    />
+                  ) : (
+                    <input name="medium" type="radio" value={medium.id} />
+                  )}
+                  {' '}
+                  {mediumFormatName(medium)}
+                  {' '}
+                  {medium.position}
+                  {nonEmpty(medium.name) ? (
+                    <>
+                      {': '}
+                      {medium.name}
+                    </>
+                  ) : null}
```

Comment:
Should `addColonText` be used here somehow?

Reply:
I guess we could construct a `const mediumHeading = mediumFormatName(medium) + ' ' + medium.position` string first and then do:
```
                  {nonEmpty(medium.name) ? (
                    <>
                      {addColonText(mediumHeading)}
                      {' '}
                      {medium.name}
                    </>
                  ) : mediumHeading}

```

Does that seem preferable?

Reply:
If it is about i18n support:
- `root/static/scripts/common/components/MediumDescription.js` already has a message `{medium_format} {position}` that can likely be reused for `mediumHeading`.
- A similar message can probably be added for the quoted code: `{medium_heading}: {name}`.

Reply:
Followed @yvanzo's suggestion here :)